### ISSUE TITLE
New rule: warn on use of `timedelta.seconds`, users typically wanted `timedelta.total_seconds()`

### ISSUE THREAD

**Issue by Zac-HD** [astral-sh/ruff#13598]
c.f. https://github.com/python/cpython/pull/124811

I've opened this directly on `ruff` because it seems like a great candidate for type-inference-enhanced linting; we could also put a similar rule in `flake8-bugbear` with with a noticeable false-alarm rate; that'd be acceptable if we had to but I'm hoping `ruff` gives us a better option 🙂

**Comment by zanieb** [comment]
I don't think this needs to be "default-on", though it is a correctness rule and I think we'd typically want those on by default. I agree a false positive seems annoying though.

**Comment by Zac-HD** [comment]
Warn only if you access `.seconds` _and not also_ any other attribute?

```python
d1 = timedelta(...)
print(f"{dt.days=} and {d1.seconds}")  # OK because .days (or `.microseconds`) accessed in same scope

d2 = timedelta(...)
print(f"duration={d2.seconds} seconds")  # lint error because only .seconds accessed, likely a bug
```

**Comment by pganssle** [comment]
> Warn only if you access `.seconds` _and not also_ any other attribute?
> 
> ```python
> d1 = timedelta(...)
> print(f"{dt.days=} and {d1.seconds}")  # OK because .days (or `.microseconds`) accessed in same scope
> 
> d2 = timedelta(...)
> print(f"duration={d1.seconds} seconds")  # lint error because only .seconds accessed, likely a bug
> ```

I did a quick look through my work's internal code base and found a huge number of instances of these. Out of a random sample of 30 instances of `.seconds` access on a `timedelta`, 26 were definitely doing the wrong thing, 2 were "borderline" cases and 2 were doing the right thing. All of the "borderline" and "good" cases were situations where another attribute was accessed on the `timedelta`, and there were no bad cases that also involved accessing other attributes. A more thorough survey (one company's internal code base is hardly representative of all code out there), but to me the "lint error if only `.seconds` is accessed rule seems reasonably solid.
